src/image-button/image-button.tsx

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { ButtonRef } from "../button/types";
+import { Main } from "./image-button.style";
+import { ImageButtonProps } from "./types";
+import { Text } from "../text/text";
+import { ImageWithFallback } from "src/shared/image-with-fallback/image-with-fallback";
+
+const Component = (
+    { children, src, ...otherProps }: ImageButtonProps,
+    ref: ButtonRef
+) => {
+    return (
+        <Main ref={ref} {...otherProps}>
+            <ImageWithFallback src={src} />
+            <Text.H4>{children}</Text.H4>
```

Comment:
on further thought, it's best not to nest headers in buttons. you can apply the text style on the button level instead (i.e. without the `h4 {...}`)
```suggestion
            {children}
```